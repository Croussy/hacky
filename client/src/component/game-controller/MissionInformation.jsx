import React from 'react';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faCheckCircle} from "@fortawesome/free-solid-svg-icons";

const MissionInformation = ({mission}) => {
    const isDisabledClass = !mission.isActive ? 'is-disabled' : ''
    const achievedIcon = mission.isAchieved ? <FontAwesomeIcon className={'mission-achieved-icon'} icon={faCheckCircle}/> : ''
    return (
        <div className={`mission-information ${isDisabledClass}`}>
            <div className={'mission-information-content'}>
                <div className={'mission-name'}>{mission.name}</div>
                <div className={'mission-description'}>
                    {mission.description}
                </div>
            </div>
            {achievedIcon}
        </div>
    );
};

export default MissionInformation;