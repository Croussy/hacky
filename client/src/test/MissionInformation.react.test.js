import React from "react";
import renderer from "react-test-renderer";
import MissionInformation from "../component/game-controller/MissionInformation";

const missionData = {
    name: "Test1",
    description: "Ceci permet de tester le componentInformation",
    isActive : true,
    isAchieved: false
}

test('MissionInformation component change class when is not disabled', () => {
    missionData.isActive = true
    const component = renderer.create(
        <MissionInformation mission={missionData}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})

test('MissionInformation component change class when is disabled', () => {
    missionData.isActive = false
    const component = renderer.create(
        <MissionInformation mission={missionData}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})
test('MissionInformation component change class when is achieved', () => {
    missionData.isActive = true
    missionData.isAchieved = true
    const component = renderer.create(
        <MissionInformation mission={missionData}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})