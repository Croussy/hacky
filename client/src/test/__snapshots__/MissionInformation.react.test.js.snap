// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionInformation component change class when is achieved 1`] = `
<div
  className="mission-information "
>
  <div
    className="mission-information-content"
  >
    <div
      className="mission-name"
    >
      Test1
    </div>
    <div
      className="mission-description"
    >
      Ceci permet de tester le componentInformation
    </div>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-circle-check mission-achieved-icon"
    data-icon="circle-check"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM371.8 211.8C382.7 200.9 382.7 183.1 371.8 172.2C360.9 161.3 343.1 161.3 332.2 172.2L224 280.4L179.8 236.2C168.9 225.3 151.1 225.3 140.2 236.2C129.3 247.1 129.3 264.9 140.2 275.8L204.2 339.8C215.1 350.7 232.9 350.7 243.8 339.8L371.8 211.8z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`MissionInformation component change class when is disabled 1`] = `
<div
  className="mission-information is-disabled"
>
  <div
    className="mission-information-content"
  >
    <div
      className="mission-name"
    >
      Test1
    </div>
    <div
      className="mission-description"
    >
      Ceci permet de tester le componentInformation
    </div>
  </div>
  
</div>
`;

exports[`MissionInformation component change class when is not disabled 1`] = `
<div
  className="mission-information "
>
  <div
    className="mission-information-content"
  >
    <div
      className="mission-name"
    >
      Test1
    </div>
    <div
      className="mission-description"
    >
      Ceci permet de tester le componentInformation
    </div>
  </div>
  
</div>
`;
